kuma/authkeys/tests/test_views.py

```diff
@@ -135,3 +146,52 @@ def test_delete_key(self):
         ok_(302, resp.status_code)
 
         eq_(0, Key.objects.filter(pk=self.key1.pk).count())
+
+
+class KeyViewsPermissionTest(TestCase):
+
+    def setUp(self):
+        username = 'tester23'
+        password = 'trustno1'
+        email = 'tester23@example.com'
+
+        self.user = user(username=username, email=email,
+                         password=password, save=True)
+        self.client.login(username=username, password=password)
+
+    def test_new_key_requires_permission(self):
+        url = reverse('authkeys.new', locale='en-US')
+        resp = self.client.get(url)
+        eq_(403, resp.status_code)
+
+        perm = Permission.objects.get(codename='add_key')
+        self.user.user_permissions.add(perm)
+        _cache_bust_user_perms(self)
+
+        resp = self.client.get(url)
+        eq_(200, resp.status_code)
+
+    def test_delete_key_requires_separate_permission(self):
+        self.key1 = Key(user=self.user, description='Test Key 1')
+        self.key1.save()
+
+        url = reverse('authkeys.delete', locale='en-US', args=(self.key1.pk,))
+        resp = self.client.get(url)
+        eq_(403, resp.status_code)
+        _cache_bust_user_perms(self)
+
+        resp = self.client.get(url)
+        eq_(403, resp.status_code)
+
+        perm = Permission.objects.get(codename='delete_key')
+        self.user.user_permissions.add(perm)
+        _cache_bust_user_perms(self)
+
+        resp = self.client.get(url)
+        eq_(200, resp.status_code)
+
+
+def _cache_bust_user_perms(self):
+    # Need to cache-bust the user perms by re-fetching from DB
+    # https://docs.djangoproject.com/en/1.7/topics/auth/default/#permissions-and-authorization
+    self.user = User.objects.get(username=self.user.username)
```

Comment:
No need to put this in a pseudo private function, just put it in a base test class and inherit from it in both tests.